django/forms/formsets.py

```diff
@@ -450,23 +455,28 @@ def media(self):
         else:
             return self.empty_form.media
 
+    def get_context(self):
+        return {
+            'formset': self,
+        }
+
+    def render(self, template_name=None, context=None, renderer=None):
+        return mark_safe((renderer or get_default_renderer()).render(
```

Comment:
It has taken me some time to fully understand this. It looks like the `renderer` argument was added to form as part of the changes introduced when widgets we moved to the template engine. To quote from the docs:

> Specifies the renderer to use for the form. Defaults to `None` which means to use the default renderer specified by the
FORM_RENDERER setting.

Therefore there's a specific use case that folk can customise which render is used on a per form basis. I agree it would make sense to also have this ability on a per formset basis (we'd should also make the same changes to `ErrorList`).

Reply:
Yes. Effectively we should propagate the renderer passed in from formset to form to widget and errors.